packages/db/src/core/cli/commands/link/index.ts

```diff
@@ -1,36 +1,76 @@
 import type { AstroConfig } from 'astro';
+import { homedir } from 'node:os';
 import { mkdir, writeFile } from 'node:fs/promises';
 import { bgRed, cyan } from 'kleur/colors';
+import { slug } from 'github-slugger';
 import prompts from 'prompts';
 import type { Arguments } from 'yargs-parser';
 import { PROJECT_ID_FILE, getSessionIdFromFile } from '../../../tokens.js';
 import { getAstroStudioUrl } from '../../../utils.js';
 import { MISSING_SESSION_ID_ERROR } from '../../../errors.js';
+import { basename } from 'node:path';
+import ora from 'ora';
 
 export async function cmd({ flags }: { config: AstroConfig; flags: Arguments }) {
-	const linkUrl = new URL(getAstroStudioUrl() + '/auth/cli/link');
 	const sessionToken = await getSessionIdFromFile();
 	if (!sessionToken) {
 		console.error(MISSING_SESSION_ID_ERROR);
 		process.exit(1);
 	}
-	let body = { id: flags._[4] } as {
-		id?: string;
-		projectIdName?: string;
-		workspaceIdName?: string;
-	};
-	if (!body.id) {
-		const workspaceIdName = await promptWorkspaceName();
-		const projectIdName = await promptProjectName();
-		body = { projectIdName, workspaceIdName };
+	await promptBegin();
+	const isLinkExisting = await promptLinkExisting();
+	if (isLinkExisting) {
+		const projectId = await promptExistingProjectName();
+		return await linkProject(projectId);
+	}
+
+	const isLinkNew = await promptLinkNew();
+	if (isLinkNew) {
+		const newProjectName = await promptNewProjectName();
+		const newProjectRegion = await promptNewProjectRegion();
+		const newProjectId = await createNewProject({name: newProjectName, region: newProjectRegion});
+		// TODO(fks): Actually listen for project creation before continuing
+		// This is just a dumb spinner that roughly matches database creation time.
+		const spinner = ora('Creating new project...');
+		await new Promise((r) => setTimeout(r, 4000));
```

Comment:
"First make it work, then make it good."
~ Confucius, probably

Reply:
When this is addressed, it would be great to provide a link to your project dashboard!

Reply:
oh nice, that's a great call. Just refactored the code a bit to make this even easier in the future